### PR TITLE
Fix code cov for AddressLookupStep

### PR DESCRIPTION
**Pull request by alexwakeman**
Adds extra tests to ensure code cov for newer code.

### CODE REVIEW

File: app/components/AddressLookupStep/index.js [hunk 1]
```diff
@@ -29,19 +29,24 @@ module.exports = class AddressLookupStep extends ValidationStep {
   }
 
   hasSessionSelectedAddress(session) {
-    const hasPostcodeLookup = session.postcodeLookup && session.postcodeLookup.addresses && session.postcodeLookup.selectAddressIndex;
+    const currentSession = session[this.schemaScope];
+    const hasPostcodeLookup = !([
```
Comment (jvictorcf) [inline app/components/AddressLookupStep/index.js]: Granted, I'm not a JS guy, but wouldn't this read more easily if it was a private method?
If this is a common JS way of asserting this, I'm ok with that.

File: sonar-project.properties [hunk 2]
```diff
@@ -5,7 +5,7 @@ sonar.projectVersion=0.0.1
 sonar.sources=app
 sonar.tests=test
 sonar.language=js
-sonar.exclusions=app/assets/**, app/**/*.test.js, test/**, **/healthcheck.js, app/steps/save-resume/confirm-remove-saved-application/index.js, app/middleware/draftPetitionStoreMiddleware.js, app/steps/sitemap/client.js, app/steps/error-400/template.html, app/steps/error-404/template.html, app/views/question.html, app/steps/marriage/upload/template.html, app/steps/check-your-answers/template.html, app/steps/done-and-submitted/template.html
+sonar.exclusions=app/assets/**, app/**/*.test.js, test/**, **/healthcheck.js, app/steps/save-resume/confirm-remove-saved-application/index.js, app/middleware/draftPetitionStoreMiddleware.js, app/steps/sitemap/client.js, app/steps/error-400/template.html, app/steps/error-404/template.html, app/views/question.html, app/steps/marriage/upload/template.html, app/steps/check-your-answers/template.html, app/steps/done-and-submitted/template.html, app/components/AddressLookupStep/mocks/postcodeInfo.js
```
Comment (jvictorcf) [inline sonar-project.properties]: Why are we excluding this file?

File: app/components/AddressLookupStep/index.js (hunk 1 above)
Comment (alexwakeman) [reply app/components/AddressLookupStep/index.js]: this is a known way of asserting all values are or are not with a certain value - we need this becuase `currentSession.selectAddressIndex` is often `0` which is `falsey` in JS - so standard truthy test won't work

https://stackoverflow.com/a/43635855/239036 reference.

File: sonar-project.properties (hunk 2 above)
Comment (alexwakeman) [reply sonar-project.properties]: it's only a test support file, not app code - we don't need coverage testing on it.

File: app/components/AddressLookupStep/index.js (hunk 1 above)
Comment (jvictorcf) [reply app/components/AddressLookupStep/index.js]: Got you. Thanks.

**Pull request merged**